crates/cxx-qt-lib/src/types/qstring.rs

```diff
@@ -3,52 +3,111 @@
         include!("cxx-qt-lib/include/qt_types.h");
 
-        type QString;
+        type QString = super::QString;
+    }
 
-        #[namespace = "rust::cxxqtlib1"]
-        #[rust_name = "qstring_to_rust_string"]
-        fn qstringToRustString(string: &QString) -> String;
+    #[namespace = "rust::cxxqtlib1"]
+    unsafe extern "C++" {
+        #[doc(hidden)]
+        #[rust_name = "qstring_drop"]
+        fn qstringDrop(string: &mut QString);
 
-        #[namespace = "rust::cxxqtlib1"]
+        #[doc(hidden)]
+        #[rust_name = "qstring_init_default"]
+        fn qstringInitDefault() -> QString;
+        #[doc(hidden)]
         #[rust_name = "qstring_init_from_rust_string"]
-        fn qstringInitFromRustString(string: &str) -> UniquePtr<QString>;
-        #[namespace = "rust::cxxqtlib1"]
+        fn qstringInitFromRustString(string: &str) -> QString;
+        #[doc(hidden)]
         #[rust_name = "qstring_init_from_qstring"]
-        fn qstringInitFromQString(string: &QString) -> UniquePtr<QString>;
-    }
+        fn qstringInitFromQString(string: &QString) -> QString;
 
-    impl UniquePtr<QString> {}
+        #[doc(hidden)]
+        #[rust_name = "qstring_to_rust_string"]
+        fn qstringToRustString(string: &QString) -> String;
+    }
 }
 
 /// The QString class provides a Unicode character string.
 ///
-/// Note that this is the C++ representation and String or &str should be used in Rust.
-pub type QString = ffi::QString;
+/// Note that QString is a UTF-16 whereas Rust strings are a UTF-8
+pub struct QString {
+    /// The layout has changed between Qt 5 and Qt 6
+    ///
+    /// Qt5 QString has one pointer as a member
+    /// Qt6 QString has one member, which contains two pointers and a size_t
+    #[cfg(qt_version_major = "5")]
+    _space: MaybeUninit<usize>,
+    #[cfg(qt_version_major = "6")]
+    _space: MaybeUninit<[usize; 3]>,
+}
 
-impl QString {
-    pub fn from_ref(string: &QString) -> cxx::UniquePtr<Self> {
-        ffi::qstring_init_from_qstring(string)
+impl Clone for QString {
+    /// Constructs a copy of other.
+    ///
+    /// This operation takes constant time, because QString is implicitly shared.
+    /// This makes returning a QString from a function very fast.
+    /// If a shared instance is modified, it will be copied (copy-on-write), and that takes linear time.
+    fn clone(&self) -> Self {
+        ffi::qstring_init_from_qstring(self)
     }
+}
 
-    pub fn from_str(str: &str) -> cxx::UniquePtr<Self> {
-        ffi::qstring_init_from_rust_string(str)
+impl Default for QString {
+    /// Constructs an invalid color. An invalid color is a color that is not properly set up for the underlying window system.
+    ///
+    /// The alpha value of an invalid color is unspecified.
+    fn default() -> Self {
+        ffi::qstring_init_default()
     }
 }
 
 impl std::fmt::Display for QString {
+    /// Convert the QString to a Rust string
+    ///
+    /// Note that this converts from UTF-16 to UTF-8
```

Comment:
Why was the QString -> String conversion done in `impl Display for QString` again?
Shouldn't we rather add a `Into<String>` implementation for QString and use that here?

Reply:
This allows for a `to_string()` method on the `QString` which feels quite natural? Eg `qstring.to_string()`

Reply:
Oh you mean in the impl why add a Into<String> and reuse that here... good idea i'll add it :-)

Reply:
Done